docs/Reference-for-template.json.md

```diff
@@ -52,31 +52,275 @@ You can find a sample of a complex `template.json` file here [Sample](https://gi
+- `nameLower`: replaces `John Smith (l)` with value of `ownername` parameter in lower case
+ 
+```json
+"symbols":{
+  "myconstant": {
+    "type": "generated",
+    "generator": "constant",
+    "parameters": {
+      "value":"5001"
+    },
+    "replaces":"1234"
+  },
+  "ownername":{
+    "type": "parameter",
+    "datatype":"text",
+    "replaces": "John Smith (a)",
+    "defaultValue": "John Doe"
+  },
+  "nameUpper":{
+    "type": "generated",
+    "generator": "casing",
+    "parameters": {
+      "source":"ownername",
+      "toLower": false
+    },
+    "replaces":"John Smith (U)"
+  },
+  "nameLower":{
+    "type": "generated",
+    "generator": "casing",
+    "parameters": {
+      "source":"ownername",
+      "toLower": true
+    },
+    "replaces":"John Smith (l)"
+  }
+}
+```
+
+#### Computed symbol
+
+A symbol for which the config provides a Boolean predicate whose evaluation result is the computed symbol result.
+
+|Name|Description|
+|---|---|
+|`type`|`computed`|
+|`value`| Boolean expresion to be computed.|
+|`evaluator`|Language to be used for evaluation of expression: <br />- `C++2` - default<br />- `C++`<br />- `MSBUILD`<br />- `VB`|
+
+##### Example
+
+Values of `OrganizationalAuth`, `WindowsAuth`, `MultiOrgAuth`, `SingleOrgAuth`, `IndividualAuth`, `NoAuth`, `RequiresHttps` symbols are computed based on `auth` symbol.
+
+```json
+  "symbols": {
+    "auth": {
+      "type": "parameter",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "None",
+          "description": "No authentication"
+        },
+        {
+          "choice": "Individual",
+          "description": "Individual authentication"
+        },
+        {
+          "choice": "SingleOrg",
+          "description": "Organizational authentication for a single tenant"
+        },
+        {
+          "choice": "MultiOrg",
+          "description": "Organizational authentication for multiple tenants"
+        },
+        {
+          "choice": "Windows",
+          "description": "Windows authentication"
+        }
+      ],
+      "defaultValue": "Individual",
+      "description": "The type of authentication to use"
+    },
+ "OrganizationalAuth": {
+      "type": "computed",
+      "value": "(auth == \"SingleOrg\" || auth == \"MultiOrg\")"
+    },
+    "WindowsAuth": {
+      "type": "computed",
+      "value": "(auth == \"Windows\")"
+    },
+    "MultiOrgAuth": {
+      "type": "computed",
+      "value": "(auth == \"MultiOrg\")"
+    },
+    "SingleOrgAuth": {
+      "type": "computed",
+      "value": "(auth == \"SingleOrg\")"
+    },
+    "IndividualAuth": {
+      "type": "computed",
+      "value": "(auth == \"Individual\")"
+    },
+    "NoAuth": {
+      "type": "computed",
+      "value": "(!(IndividualAuth || MultiOrgAuth || SingleOrgAuth || WindowsAuth))"
+    },
+    "RequiresHttps": {
+      "type": "computed",
+      "value": "(OrganizationalAuth)"
+    },
+```
+
+#### Bind symbol
```

Comment:
@KathleenDollard do you have any information about `bind` symbols and how they can be used?
I see that they are used in existing templates for `HostIdentifier` but I'm not sure what else they can be used for.

Reply:
I do not know. Hmm. @phenning?

Reply:
Keeping description as is for now